### PR TITLE
Node messaging app: Mention expectations regarding real time functionality

### LINKED ISSUE #26525: NodeJS Messaging App: Mention expectations regarding realtime updates
### Describe your suggestion

In the current NodeJS course, there are no lessons that introduce how to manage low-latency realtime updates, be that via long polling or websockets or some other alternative. Whilst I don't think it's vital such a lesson be written and added to the course, I think it's at least important to mention in the messaging app spec that it does not expect realtime updates (I think many people will adopt a REST API approach for their own backend).

I propose mentioning that expectation (or lack thereof) and perhaps also mentioning that if after the course the learner is interested in looking into this, they might be interested in long polling or websockets via something like `socket.io`.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/nodejs-messaging-app

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### (Optional) Discord Name

Mao

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by MaoShizhong**
## Because
Methods for implementing real time functionality (e.g. polling/SSE/websockets etc.) have not been taught yet in the Node course. Unless these are later added to the curriculum, the Node messaging app ought to clarify that the user is not expected to implement any real time functionality in this project.

## This PR
- Adds mention of real time functionality expectations and explanation
- Fixes nested numbered list indentation to follow style guide
- Replaces `*` for `-` for extra credit unordered list to follow style guide

## Issue
Closes #26525

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by MaoShizhong** [comment]
Side note - I noticed that the messaging app and Where's Waldo project `.md` files are currently in `react/react_and_the_backend` and not in the `nodeJS` directory.
How straightforward would this be to restructure?  If this will not cause any significant conflicts then after this has been approved and merged, I can open a new PR to move them into the appropriate `nodeJS` directory.

**Comment by JoshDevHub** [comment]
> Side note - I noticed that the messaging app and Where's Waldo project .md files are currently in react/react_and_the_backend and not in the nodeJS directory.
How straightforward would this be to restructure? If this will not cause any significant conflicts then after this has been approved and merged, I can open a new PR to move them into the appropriate nodeJS directory.

You'll notice if you closely read the project prompts that there's nothing specific to NodeJS about them. This is partially because people who are this far can be given very high level instructions, but it's also because these same projects exist in the Ruby/Rails path. But in that path, they are placed in the React content rather than the "backend" -- owing to the way things are ordered there (backend comes before React).

Anyways, I probably don't think it makes sense to move them inside the `nodeJS/` directory when the Rails path is using them, although I get that it's probably a bit of a confusing navigational experience from a node learner's perspective.

As another aside: rails learners doing this project actually *have* been exposed to websockets :laughing: 
I think it's still fine though because although they've been shown how to use them in a Rails monolith, they haven't been shown how to interact with websockets on the React end of a split FE/BE app.

File: react/react_and_the_backend/project_messaging_app.md [hunk 1]
```diff
@@ -7,17 +7,18 @@ Think of your favorite messaging app - Discord, Telegram, WhatsApp, Facebook Mes
...
-    1. Authorization
-    1. Sending messages to another user
-    1. Customizing a user profile
+   1. Authorization
+   1. Sending messages to another user
+   1. Customizing a user profile
 1. Spend some time planning out your app. What will the user interface look like? What will the data model look like? What libraries will you need to use?
 1. Start building your app by implementing the core functionality in your backend and frontend.
+   1. You may have realised that a REST API backend cannot handle real time updates. It is "request-response", so the server can only respond to a request. If a user sends a message to another user, a REST API cannot automatically notify the recipient as the recipient will not have requested that data. Whilst there are ways to implement real time functionality (e.g. polling, server-sent events, websockets etc.), these methods have not been taught and so you are not expected to implement any in this app.
```
Comment (JoshDevHub) [inline react/react_and_the_backend/project_messaging_app.md:15]:
```suggestion
   1. You may have realized that a REST API backend cannot handle real time updates. It is "request-response", so the server can only respond to a request. If a user sends a message to another user, a REST API cannot automatically notify the recipient as the recipient will not have requested that data. Whilst there are ways to implement real time functionality (e.g. polling, server-sent events, websockets etc.), these methods have not been taught, so you are not expected to implement any in this app.
```
"realize" for "realise" is an American English vs UK English thing. The style guide indicates a preference for American English.

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
Couple incredibly small nits but otherwise this looks good.

**Comment by MaoShizhong** [comment]
Good catch, I need to remember to switch to AE mode when working on these repos.
I have reworded the last bit of the additions to be a bit more pathway-agnostic whilst hopefully still retaining the same message.

**Review by JoshDevHub — APPROVED** [review]
Awesome work @MaoShizhong :raised_hands:

**Pull request merged**